### PR TITLE
ci: upload cli version list

### PR DESCRIPTION
**Pull request by msanft**
### Proposed change(s)

- Upload a version list of CLI versions to the versionsapi as a CI action
- [Successful run](https://github.com/edgelesssys/constellation/actions/runs/4363889664/jobs/7631996913)

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: internal/versionsapi/cli/add.go [hunk 1]
```diff
@@ -230,6 +232,19 @@ func parseAddFlags(cmd *cobra.Command) (addFlags, error) {
...
+	}
+	var kind versionsapi.VersionKind
+	switch kindFlag {
+	case "image":
+		kind = versionsapi.VersionKindImage
+	case "cli":
+		kind = versionsapi.VersionKindCLI
+	default:
+		return addFlags{}, fmt.Errorf("invalid kind %q", kind)
+	}
```
Comment (katexochen) [inline internal/versionsapi/cli/add.go:247]: Use `versionsapi.VersionKindFromString()` instead.

File: internal/versionsapi/list.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func (l List) ValidateRequest() error {
 	if l.Granularity != GranularityMajor && l.Granularity != GranularityMinor {
 		retErr = errors.Join(retErr, fmt.Errorf("granularity %q is not supported", l.Granularity))
 	}
-	if l.Kind != VersionKindImage {
+	if l.Kind != VersionKindImage && l.Kind != VersionKindCLI {
```
Comment (katexochen) [inline internal/versionsapi/list.go:77]: Maybe switch to `l.Kind == VersionKindUnknown`?

File: .github/workflows/on-release.yml [hunk 3]
```diff
@@ -67,12 +67,24 @@ jobs:
...
+      - name: Add CLI version to versionsapi
+        working-directory: internal/versionsapi/cli
+        run: |
+          latest_flag=$([[ "${LATEST}" = "true" ]] && echo "--latest" || echo "")
+          ./versionsapi add \
+            --release \
+            --stream "stable" \
+            --kind "cli" \
             --version "${TAG}" \
             "${latest_flag}"
```
Comment (katexochen) [inline .github/workflows/on-release.yml:90]: Could you try to replace this with the according workflow call? Can't remember why it isn't used her, and @malt3 can't either. It is currently unsynchronized, so things could easily go wrong.

**Review by katexochen — CHANGES_REQUESTED** [review]

**Comment by msanft** [comment]
[Successful run after changes](https://github.com/edgelesssys/constellation/actions/runs/4373371728/jobs/7652321337)

**Comment by malt3** [comment]
Just a heads-up: This branch should be rebased before merge and bazel build files should be generated.
[This document](https://github.com/edgelesssys/constellation/blob/main/dev-docs/workflows/build-test-run.md) has up-to-date information on the required tasks (`bazel run gazelle`).

**Review by katexochen — APPROVED** [review]
lgtm

**Pull request merged**